doc/events/http_failure.md

```diff
@@ -0,0 +1,30 @@
+---
+module: [kind=event] http_failure
+see: http.request To send an HTTP request.
+---
+
+The @{http_failure} event is fired when an HTTP request fails.
+
+This event is normally handled inside @{http.get} and @{http.post}, but it can still be seen when using @{http.request}.
+
+## Return Values
+1. @{string}: The URL of the site requested.
+2. @{string}: An error describing the failure.
+3. @{http.Response|nil}: A response handle if the connection succeeded, but the server's response indicated failure.
+
+## Example
+Prints an error why the website cannot be contacted:
+```lua
+http.request("http://this.website.does.not.exist")
+local event, url, err = os.pullEvent("http_failure")
```

Comment:
As with `alarm`, these should definitely be checking the URL.

_Maybe_ worth using a `tweaked.cc`/`computercraft.cc` domain instead of httpbin? On the basis that I can probably commit to those being maintained for as long as this mod is :). Though I guess httpbin isn't probably going to vanish any time soon.